R/merge_sce_list.R

```diff
@@ -0,0 +1,200 @@
+#'
+#' This function takes an optionally-named (if named, ideally by a form of
+#'  library ID) list of SingleCellExperiment (SCE) objects and merges them into
+#'  one SCE object. At least some genes must be present in all SCEs in order to
+#'  merge them. Currently any present altExps are not retained.
+#'
+#'  Original SCE contents are modified or retained as follows:
+#'  - The resulting colData slot will include a new column specified by
+#'    `batch_column` (default "library_id") that either holds the originating SCE
+#'    object's name (referred to as `sce_name` here), of if it is unnamed then
+#'    its index in the provided `sce_list`.
+#'  - The resulting colData slot will include another new column `cell_id_column`
+#'    (default "cell_id") that is formatted as `{sce_name}-{rowname}`, where
+#'    `sce_name` is the given batch (library) value and `rowname` is the colData
+#'    row's rowname which often, but not always, holds a unique cell barcode.
+#'    This column serves to match cells back to both originating batch and cell name.
+#'  - The resulting colData rownames will be updated to match `cell_id`.
+#'  - The resulting rowData slot column names will be appended with the given
+#'    SCE's name, as `{column_name}-{sce_name}` except for columns whose names
+#'    are indicated to preserve with the `preserve_rowdata_cols` argument.
+#'  - The resulting rowData slot rownames will be updated to `{rowname}-{sce_name}`.
+#'
+#'
+#' @param sce_list A list of SingleCellExperiment objects. The list may optionally
+#'  be named with batch information. If no names are provided, names will be generated
+#'  based on the SCE's index.
+#' @param batch_column A character value giving the resulting colData column name
+#'  to differentiate originating SingleCellExperiment objects. Often these values
+#'  are library IDs. Default value is `library_id`.
+#'  @param cell_id_column A character value giving the resulting colData colum name
+#'  to hold unique cell IDs formatted as their batch and original rowname. Default
+#'  value is `cell_id`.
+#' @param retain_coldata_cols A vector of colData columns which should be retained
+#'  in the the final merged SCE object.
+#' @param preserve_rowdata_cols A vector of column names that appear in originating
+#'  SCE objects' rowData slots which should not be renamed with
+#'  the given SCE object name or index is name is not given. These are generally
+#'  columns which are not specific to the given library's preparation or statistics.
+#'  For example, such a vector might contain items like "Gene", "ensembl_ids", etc.
+#'
+#' @return A SingleCellExperiment object containing all SingleCellExperiment objects
+#'   present in the inputted list
+#' @export
+#'
+#' @import SingleCellExperiment
+merge_sce_list <- function(sce_list = list(),
+                           batch_column = "library_id",
+                           cell_id_column = "cell_id",
```

Comment:
My concern is if they, by chance, already have something named `cell_id` we'd overwrite their data. That said "partly because it seems to be the norm for SCE functions" is a solid argument.

I might pop it at the end of options, then.